### ISSUE TITLE
.NET 7 support

### ISSUE THREAD

**Issue by mbp** [erikbra/grate#280]
Running "grate" within a Docker container `mcr.microsoft.com/dotnet/runtime:7.0`, outputs the following error message:

```
You must install or update .NET to run this application.
App: /opt/bin/grate
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)
.NET location: /usr/share/dotnet

The following frameworks were found:
7.0.0 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```

Is it possible to support system where only .NET 7 is installed?

**Comment by wokket** [comment]
I'd love to upgrade to net7 as soon as practicable, but there's obviously going to be people that are still running in net6, and can't upgrade.

What's slightly more annoying is we package and self-contain to avoid this sort of issue (ie the net6 dependencies should be baked in), that's apparently not working here, _but_ we also have bugs where the trimmed image is cause issues, so we're copping the worst of both worlds, and gaining the advantages of neither.

**Comment by erikbra** [comment]
This _should_ work with the self-contained version of grate, @mbp . Which binary are you running in the example above?

**Comment by mbp** [comment]
Hi @erikbra I am using a Dockerfile like this:

```
FROM mcr.microsoft.com/dotnet/sdk:7.0 AS build-env
WORKDIR /app

RUN dotnet tool install -g grate

FROM mcr.microsoft.com/dotnet/runtime:7.0

COPY --from=build-env /root/.dotnet/tools/ /opt/bin
ENV PATH="/opt/bin:${PATH}"

ENTRYPOINT ["grate", "-h"]
```

This worked with "6.0", but not with "7.0".

**Comment by dgeller-OUHSC** [comment]
I just ran across the problem in WSL:


```
dubs@L5CD0254Q09:/mnt/c/Users/daryl.geller/code/project/src/YH$ dotnet tool install grate -g
You can invoke the tool using the following command: grate
Tool 'grate' (version '1.4.0') was successfully installed.
dubs@L5CD0254Q09:/mnt/c/Users/daryl.geller/code/project/src/YH$ grate
You must install or update .NET to run this application.

App: /home/dubs/.dotnet/tools/grate
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)
.NET location: /usr/share/dotnet

The following frameworks were found:
  7.0.1 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Learn about framework resolution:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=ubuntu.22.04-x64
dubs@L5CD0254Q09:/mnt/c/Users/daryl.geller/code/project/src/YH$ dotnet --list-runtimes
Microsoft.AspNetCore.App 6.0.11 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 7.0.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 7.0.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
dubs@L5CD0254Q09:/mnt/c/Users/daryl.geller/code/project/src/YH$

```

**Comment by erikbra** [comment]
Hi, @mbp. This should work if one sets the dotnet's roll forward parameter. It can be set in a couple of ways, including an environment variable, see 

https://learn.microsoft.com/en-gb/dotnet/core/runtime-discovery/troubleshoot-app-launch?pivots=os-windows#configure-roll-forward-behavior
https://learn.microsoft.com/en-gb/dotnet/core/tools/dotnet-environment-variables#dotnet_roll_forward
https://learn.microsoft.com/en-gb/dotnet/core/tools/dotnet#options-for-running-an-application and 

Could you please try to see if setting the environment variable `DOTNET_ROLL_FORWARD` to `LatestMajor` makes it work?

That said, you don't _have_ to have .NET installed to run grate. You can use the self-contained version, just pick the one for your architecture among the downloads here: https://github.com/erikbra/grate/releases. Then it will suffice with a standard Linux distro (or Windows, for that matter), without dotnet installed.

**Comment by mbp** [comment]
Hi @erikbra - thank you, I can confirm the workaround works. I will use it.

I will consider to use the self-contained version in the future. However on developer machines I do find it easier to use `dotnet tool install -g grate`, but that is now broken by default in current net7.0 version.

I don't know if it would be possible to multi-target the dotnet tool so both net6.0 and net7.0 builds are available? I mean changing `<TargetFramework>net6.0</TargetFramework>` to `<TargetFrameworks>net6.0;net7.0</TargetFrameworks>`

**Comment by wokket** [comment]
I'm also copping this in a CI scenario, we `dotnet tool install grate -g --version 1.3.2`, and get the following from teh AzDo build agent:

```
You can invoke the tool using the following command: grate
Tool 'grate' (version '1.3.2') was successfully installed.
You must install or update .NET to run this application.

App: C:\Users\VssAdministrator\.dotnet\tools\grate.exe
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)
.NET location: C:\hostedtoolcache\windows\dotnet

The following frameworks were found:
  3.1.32 at [C:\hostedtoolcache\windows\dotnet\shared\Microsoft.NETCore.App]
  7.0.3 at [C:\hostedtoolcache\windows\dotnet\shared\Microsoft.NETCore.App]

Learn about framework resolution:
[https://aka.ms/dotnet/app-launch-failed](https://mcas-proxyweb.mcas.ms/certificate-checker?login=false&originalUrl=https%3A%2F%2Faka.ms.mcas.ms%2Fdotnet%2Fapp-launch-failed%3FMcasTsid%3D26110&McasCSRF=b6abb2ce215c8846955286dc7315439de6a81a956fbedcffbfbc1205e84e91e2)

To install missing framework, download:
[https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=win10-x64](https://mcas-proxyweb.mcas.ms/certificate-checker?login=false&originalUrl=https%3A%2F%2Faka.ms.mcas.ms%2Fdotnet-core-applaunch%3Fframework%3DMicrosoft.NETCore.App%26framework_version%3D6.0.0%26arch%3Dx64%26rid%3Dwin10-x64%26McasTsid%3D26110&McasCSRF=b6abb2ce215c8846955286dc7315439de6a81a956fbedcffbfbc1205e84e91e2)
```

**Comment by erikbra** [comment]
Fixed by #313
